src/dataflow/components/tools/dataflow-tool.tsx

```diff
@@ -98,21 +104,22 @@ export default class DataflowToolComponent extends BaseComponent<IProps, IState>
       documentContent.tileMap.forEach(tile => {
         if (tile.content.type === "Dataflow") {
           const programContent = tile.content as DataflowContentModelType;
-          programContent.setProgramRunId(id);
-          programContent.setProgramStartEndTime(startTime, endTime);
-          programContent.setRunningStatus(endTime);
+          programContent.setProgramRunId(startParams.runId);
+          programContent.setProgramStartEndTime(startParams.startTime, startParams.endTime);
+          programContent.setRunningStatus(startParams.endTime);
         }
       });
-      const properties: IDocumentProperties = { dfProgramId: document.key, dfRunId: id };
+      const properties: IDocumentProperties = { dfProgramId: document.key, dfRunId: startParams.runId };
       if (document.title) properties.originTitle = document.title;
-      if (datasetName.length > 0) properties.dfHasData = "true";
+      if (startParams.title.length > 0) properties.dfHasData = "true";
+      if (startParams.hasRelay) properties.dfHasRelay = "true";
       // create and load the new document
-      const params: ICreateOtherDocumentParams = {
-              title: datasetName || `${document.title}-${getLocalTimeStamp(Date.now())}`,
+      const createParams: ICreateOtherDocumentParams = {
+              title: startParams.title || `${document.title}-${getLocalTimeStamp(Date.now())}`,
```

Comment:
Not sure if now is the time to fix it, but if the user changes the name of a relay-only program, the user-specified name is not used.